presto-main/src/main/java/com/facebook/presto/execution/SqlStageExecution.java

```diff
@@ -542,7 +541,7 @@ public void recordGetSplitTime(long start)
     private static Split createRemoteSplitFor(TaskId taskId, URI remoteSourceTaskLocation, TaskId remoteSourceTaskId)
     {
         // Fetch the results from the buffer assigned to the task based on id
-        URI splitLocation = uriBuilderFrom(remoteSourceTaskLocation).appendPath("results").appendPath(String.valueOf(taskId.getId())).build();
+        String splitLocation = remoteSourceTaskLocation.toASCIIString() + "/results/" + taskId.getId();
```

Comment:
what are other string options except ASCII String? :P

Reply:
It has a `toString` as well. The only difference is that `toASCIIString` will also encode the result string when there is non-ASCII character in the string. Do you have a preference? :)